ansible/roles/build-ossec-deb-pkg/library/ossec_urls.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+DOCUMENTATION = '''
+---
+module: ossec_urls
+short_description: Gather facts for OSSEC download URLs
+description:
+  - Gather version, checksum, and URL info for OSSEC downloads
+author:
+    - Conor Schaefer (@conorsch)
+    - Freedom of the Press Foundation (@freedomofpress)
+requirements:
+    - requests
+options:
+  ossec_version:
+    description:
+      - version number of release to download
+    default: "2.8.2"
+    required: no
+notes:
+  - The OSSEC version to download is hardcoded to avoid surprises.
+    If you want a newer version than the current default, you should
+    pass the version in via I(ossec_version).
+'''
+EXAMPLES = '''
+- ossec_urls:
+    ossec_version: "2.8.2"
+'''
+
+from StringIO import StringIO
+from urlparse import urljoin
+import re
+
+HAS_REQUESTS = True
+try:
+    import requests
+except ImportError:
+    HAS_REQUESTS = False
+
+
+
+class OSSECURLs():
+
+    def __init__(self, ossec_version):
+        self.ossec_version = ossec_version
+
+        checksums = self.parse_checksums()
+
+        self.ansible_facts = dict(
+            ossec_version=self.ossec_version,
+            ossec_tarball_filename=self.ossec_tarball_filename,
+            ossec_tarball_url=self.ossec_tarball_url,
+            ossec_checksum_filename=self.ossec_checksum_filename,
+            ossec_checksum_url=self.ossec_checksum_url,
+            )
+
+        self.ansible_facts.update(checksums)
+
+    @property
+    def ossec_tarball_filename(self):
+        return "ossec-hids-{}.tar.gz".format(self.ossec_version)
+
+
+    @property
+    def ossec_tarball_url(self):
+        return "https://github.com/ossec/ossec-hids/archive/{}.tar.gz".format(
+                self.ossec_version)
+
+
+    @property
+    def ossec_checksum_url(self):
+        return "https://github.com/ossec/ossec-hids/releases/download/{}/{}".format(
+                self.ossec_version, self.ossec_checksum_filename)
+
+
+    @property
+    def ossec_checksum_filename(self):
+        return "{}-checksum.txt".format(self.ossec_tarball_filename)
+
+
+    def parse_checksums(self):
+        r = requests.get(self.ossec_checksum_url)
+        checksum_regex = re.compile(r'''
+                                    ^MD5\(
+                                    '''
+                                    +re.escape(self.ossec_tarball_filename)+
+                                    r'''\)=\s+(?P<ossec_md5_checksum>[0-9a-f]{32})\s+
+                                    SHA1\(
+                                    '''
+                                    +re.escape(self.ossec_tarball_filename)+
+                                    r'''\)=\s+(?P<ossec_sha1_checksum>[0-9a-f]{40})$
+                                    ''', re.VERBOSE | re.MULTILINE
+                                    )
```

Comment:
This regex is a little fragile (depends _exactly_ on the format of that checksums document from OSSEC) and proably also overly complex, but I think it's ok for now.